tests/Oryx.BuildImage.Tests/Ruby/RubySampleAppsTest.cs

```diff
@@ -0,0 +1,129 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Oryx.BuildScriptGenerator.Common;
+using Microsoft.Oryx.Tests.Common;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Oryx.BuildImage.Tests
+{
+    [Trait("platform", "ruby")]
+    public class RubySampleAppsTest : SampleAppsTestBase
+    {
+        public RubySampleAppsTest(ITestOutputHelper output) : base(output)
+        {
+        }
+        private DockerVolume CreateSampleAppVolume(string sampleAppName) =>
+            DockerVolume.CreateMirror(Path.Combine(_hostSamplesDir, "ruby", sampleAppName));
+
+        [Fact]
+        public void GeneratesScript_AndBuildSinatraApp()
+        {
+            // Arrange
+            var appName = "sinatra-app";
+            var volume = CreateSampleAppVolume(appName);
+            var appDir = volume.ContainerDir;
+            var appOutputDir = "/tmp/app-output";
+            var script = new ShellScriptBuilder()
+                .SetEnvironmentVariable(
+                    SdkStorageConstants.SdkStorageBaseUrlKeyName,
+                    SdkStorageConstants.DevSdkStorageBaseUrl)
+                .AddBuildCommand($"{appDir} -o {appOutputDir}")
+                .ToString();
+
+            // Act
+            var result = _dockerCli.Run(new DockerRunArguments
+            {
+                ImageId = _imageHelper.GetVsoBuildImage(),
```

Comment:
Could you add one test which uses GitHubActions image to test the dynamic install too?